src/Raven.Server/Documents/Sharding/Handlers/Processors/Documents/ShardedDocumentHandlerProcessorForGet.cs

```diff
@@ -113,15 +114,26 @@ protected override async ValueTask<DocumentsResult> GetDocumentsImplAsync(Transa
 
         if (startsWith != null)
         {
-            op = new ShardedStreamDocumentsOperation(HttpContext, changeVector, startsWith.IdPrefix, startsWith.Matches, startsWith.Exclude,
-                startsWith.StartAfterId, token);
+            op = new ShardedStreamDocumentsOperation(HttpContext, startsWith.IdPrefix, startsWith.Matches, startsWith.Exclude, startsWith.StartAfterId,
+                changeVector, token);
         }
         else // recent docs
         {
             op = new ShardedStreamDocumentsOperation(HttpContext, changeVector, token);
         }
 
         var results = await RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(op, CancellationToken);
+        if (results.Result == null)
+        {
+            Debug.Assert(results.StatusCode == (int)HttpStatusCode.NotModified);
```

Comment:
Please add a message to Debug.Assert as the second parameter

Reply:
done